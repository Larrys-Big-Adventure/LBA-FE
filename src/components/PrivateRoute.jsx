import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import { Route, withRouter } from 'react-router-dom';

const PrivateRoute = ({ component: Component, path, history, ...rest }) => {
  const [loading, setLoading] = useState(false);
  const { token } = localStorage;
  useEffect(() => {
    setLoading(true);
    const fn = () => {
      if (!token) {
        setLoading(false);
        history.push('/login');
      }
    };
    fn();
    setLoading(false);
  }, [path, history, token]);
  const render = props => (token ? <Component {...props} /> : null);
  return (
    <Route exact path={path} render={render} {...rest} loading={loading} />
  );
};

PrivateRoute.defaultProps = {
  path: '*',
  component: () => {}
};

PrivateRoute.propTypes = {
  path: PropTypes.string,
  history: PropTypes.shape({
    push: PropTypes.func.isRequired
  }).isRequired,
  component: PropTypes.func
};

export default withRouter(PrivateRoute);
