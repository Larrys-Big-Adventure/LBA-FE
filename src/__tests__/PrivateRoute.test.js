import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { cleanup } from '@testing-library/react';
import render from '../__testHelpers__/render';

import PrivateRoute from '../components/PrivateRoute.jsx';

afterEach(cleanup);

describe('PrivateRoute', () => {
  it('renders without crashing', () => {
    render(<PrivateRoute />);
  });
});
