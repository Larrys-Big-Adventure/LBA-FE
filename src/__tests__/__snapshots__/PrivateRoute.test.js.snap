// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrivateRoute matches previous snapshot 1`] = `null`;
